h1st/model/fuzzy/fuzzy_modeler.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from h1st.model.modeler import Modeler
+from h1st.model.fuzzy.fuzzy_model import FuzzyModel
+from h1st.model.fuzzy.fuzzy_variables import FuzzyVariables
+from h1st.model.fuzzy.fuzzy_rules import FuzzyRules
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class FuzzyModeler(Modeler):
+    """
+    FuzzyModeler is a Modeler for FuzzyModel. Using FuzzyVariables and FuzzyRules
+    FuzzyModeler builds ControlSystem from skfuzzy package and inject it
+    into FuzzyModel's rules property. For more information, check out
+    https://scikit-fuzzy.github.io/scikit-fuzzy/.
+
+    """
+
+    def __init__(self, model_class=FuzzyModel):
+        self.model_class = model_class
+
+    def build_model(self, variables: FuzzyVariables, rules: FuzzyRules) -> FuzzyModel:
+        """
+        Build FuzzyModel using variables and rules.
+
+        .. code-block:: python
+            :caption: example
+            vars = FuzzyVariables()
+            vars.add(
+                var_name='var1',
+                var_type='antecedent',
+                var_range=np.arange(0, 15+1e-5, 0.5),
+                membership_funcs=[('normal', fm.GAUSSIAN, [3, 3.3]),
+                                  ('abnormal', fm.TRIANGLE, [8, 15, 15])]
+            )
+            vars.add(
+                var_name='conclusion1',
+                var_type='consequent',
+                var_range=np.arange(0, 10+1e-5, 0.5),
+                membership_funcs=[('no', fm.TRAPEZOID, [0, 0, 4, 6]),
+                                  ('yes', fm.TRAPEZOID, [4, 6, 10, 10])]
+            )
+            rules = FuzzyRules()
+            rules.add_rule(
+                'rule1',
+                if_term=vars.get('var1')['abnormal'],
+                then_term=vars.get('conclusion1')['yes'])
+            modeler = FuzzyModeler()
+            model = modeler.build_model(vars, rules)
+        """
+        if not isinstance(rules, FuzzyRules):
+            raise ValueError(f"{type(rules)} is not an instance of FuzzyRules")
+        if not isinstance(variables, FuzzyVariables):
+            raise ValueError(f"{type(variables)} is not an instance of FuzzyRules")
+
+        # Build skfuzzy control system.
+        self.model_class = FuzzyModel(variables=variables, rules=rules)
```

Comment:
This is not a `model_class`. It is the model instance itself. So perhaps use `self.model`, but whatever it is, it should be consistent with the overall `Modeler` pattern.

Reply:
In fact, the `FuzzyModeler` should *not* have a reference to the model instance. It is a factory, and as such can create many different model instances. This should simply be:
```
m = FuzzyModel(variables, rules)
return m
```

This begs the question, what is `self.model_class` for? Likely answer: it is to support any subclass of FuzzyModel. In that case, you should do something like:
```
m = self.model_class(variables, rules)
return m
```

Reply:
@ctn agree. let me update to 
```
return self.model_class(variables, rules)
```